src/main.rs

```diff
@@ -6,31 +6,68 @@ use std::thread;
 use sdl2::event::{Event, WindowEventId};
 use sdl2::video::{Window, WindowPos, RESIZABLE};
 use sdl2::render::{RenderDriverIndex, SOFTWARE, Renderer, RenderDrawer};
-use sdl2::rect::Rect;
+use sdl2::rect::{Point, Rect};
 use sdl2::pixels::Color;
 use sdl2::keycode::KeyCode;
 
 use pwong::entities::paddle::{Paddle, PaddleDirection};
+use pwong::entities::ball::{Ball};
 use pwong::entities::keymap::{KeyPressMap};
 
+fn draw_circle(drawer: &mut RenderDrawer, ball: &mut Ball) {
+    drawer.set_draw_color(Color::RGB(255, 157, 0));
+    let mut f = 1 - ball.r;
+    let mut ddf_x = 1;
+    let mut ddf_y = -2 * ball.r;
+    let mut x = 0;
+    let mut y = ball.r;
+    drawer.draw_point(Point::new(ball.x, ball.y + ball.r));
```

Comment:
Wow, this seems painful.  I wonder if we'll figure out a simpler or easier-to-read version later.